gems/aws-sdk-core/lib/aws-sdk-core/auth_schemes/sigv4.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module AWS::SDK::Core
+  module AuthSchemes
+    # SigV4 authentication scheme.
+    class SigV4 < Hearth::AuthSchemes::Base
+      def initialize
+        super(
+          scheme_id: 'aws.auth#sigv4',
+          signer: Signers::SigV4.new,
```

Comment:
Signer may need to take options here, passed from initialize in auth scheme. This allows for override of signer. Signer properties may need to NOT supersede these.

Reply:
I added an option to override it.  I think its up to whatever signer implementation you provide to handle precedence of signer properties passed to it on sign.

Reply:
The way signer works now is if you pass options to sign_request, it overrides SigV4.new() options... that may be wrong now. I think the Signer options might need to override the options passed into the operation because those will come from auth options as signer params now.